src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1500,6 +1502,14 @@ public function getContent(bool $asResource = false)
         return $this->content;
     }
 
+    /**
+     * Get the application/x-form-urlencoded or application/json request body.
+     */
+    public function getPayload(): InputBag
+    {
+        return $this->payload ??= $this->request->count() ? $this->request : new InputBag($this->toArray());
```

Comment:
should we clone `$this->request` before returning it ? For JSON requests, modifying the InputBag returned from `getPayload` does not modify the Request object, while it does for form-urlencoded requests right now. This is a surprising behavior.

Reply:
I agree with cloning, and I also think we should not store the resulting payload in a property. We should return a new instance all the time. That would close the sync/duplication issues.